guides/source/configuring.md

```diff
@@ -1617,6 +1617,27 @@ You can also use `Rails::Application.config_for` to load whole configuration fil
   Rails.configuration.payment['merchant_id'] # => production_merchant_id or development_merchant_id
   ```
 
+`Rails::Application.config_for` can also merge configurations deeply. To achieve
```

Comment:
I think we need to reword "merge configurations deeply". If I was a beginner I'd have no idea what this means. 

I'd reword this to: "`Rails::Application.config_for` supports a `shared` configuration to group common configurations. The shared configuration will be merged into the environment configuration." or something similar.